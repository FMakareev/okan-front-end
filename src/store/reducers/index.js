import { combineReducers } from 'redux';
import { reducer as formReducer } from 'redux-form';
// import { localizeReducer } from 'react-localize-redux';

export const rootReducer = combineReducers({
  form: formReducer,
  // locale: localizeReducer,
});

export default rootReducer;
