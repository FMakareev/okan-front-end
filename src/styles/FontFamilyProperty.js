/**
 * @params {string} fontFamily - string name of the font name property in the theme
 * @params {object} theme
 * */
export const FontFamilyProperty = ({ fontFamily, theme }) => {
  try {
    if (fontFamily && theme.fontFamily) {
      return `font-family: '${theme.fontFamily[fontFamily]}';`;
    }
  } catch (e) {
    console.error(e);
  }
  return undefined;
};
export default FontFamilyProperty;
