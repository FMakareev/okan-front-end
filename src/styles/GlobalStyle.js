import { createGlobalStyle } from 'styled-components'


export const GlobalStyle = createGlobalStyle`
  *{
    box-sizing: border-box;
  }
  body, html, #app {
    padding: 0;
    margin: 0;
    width: 100%;
    min-height: 100vh;
    background-color: #fff;
  }
`;
