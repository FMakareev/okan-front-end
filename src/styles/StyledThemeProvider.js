import React from 'react';
import PropTypes from 'prop-types';
import { Provider } from 'rebass';

/** Variants */
import { ButtonVariant } from './variants/buttonVariant';
import { ButtonSize } from './variants/buttonSize';
import { ButtonComment } from './variants/buttonComment';
import { TextVariant } from './variants/textVariant';
import { InputSize } from '@lib/styles/variants/InputSize';
import { InputVariant } from '@lib/styles/variants/InputVariant';

export const ColorPallet = {
  color0: '#FFFFFF',
  color1: '#E5E5E5',
  color2: '#DEDEDE',
  color3: '#4F4F4F',
  color4: '#848484',
  color5: '#007FAF',
  color6: '#006699',
  color7: '#00649C',
  color8: '#007CC4',
  color9: '#F97E63',
  color10: '#AEAEAE',
  color11: '#333333',
  color12: '#DF4624',
  color13: '#f2f2f2',
  color14: '#fff3c1',
};

const Space = [
  0,
  2,
  4,
  8,
  12,
  16,
  18,
  20,
  24,
  28,
  32,
  36,
  40,
  44,
  48,
  52,
  56,
  60,
  64,
  68,
  72,
  76,
  80,
];

const Weight = [300, 500, 700];

const Breakpoints = ['1024px', '1200px', '1440px'];

const boxShadow = [
  '2px 4px 8px 0px rgba(138,138,138,0.5)',
  '2px 4px 8px 0px rgba(138,138,138,0.5)',
  '2px 2px 4px 0px rgba(0,127,175,1)',
];

const textAlign = ['left', 'center', 'right'];

export const StyledThemeProvider = ({ children }) => (
  <Provider
    theme={{
      space: Space,
      fontSizes: Space,
      fontWeight: Weight,
      lineHeight: Space,
      boxShadow,
      borderRadius: Space,
      borderColor: ColorPallet,
      breakpoints: Breakpoints,
      colors: ColorPallet,
      variant: {
        buttons: ButtonVariant,
        buttonSize: ButtonSize,
        buttonComment: ButtonComment,
        text: TextVariant,
        inputSize: InputSize,
        inputVariant: InputVariant,
      },
      fontFamily: {
        primary300: 'Museo Sans 300',
        primary500: 'Museo Sans 500',
        primary700: 'Museo Sans 700',
        secondary: 'Circe Regular',
        secondaryBold: 'Circe Bold',
      },
      textAlign: textAlign,
    }}>
    {children}
  </Provider>
);

StyledThemeProvider.propTypes = {
  children: PropTypes.element.isRequired,
};

StyledThemeProvider.defaultProps = {
  children: 'children is empty',
};

export default StyledThemeProvider;
