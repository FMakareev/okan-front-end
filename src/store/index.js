/* global isBrowser */
import { createStore, applyMiddleware } from 'redux';
import thunkMiddleware from 'redux-thunk';

import { composeWithDevTools } from 'redux-devtools-extension';
import { rootReducer } from './reducers/index';

const createStoreMiddleware = applyMiddleware(thunkMiddleware)(createStore);
const initialState = isBrowser ? window.PRELOADED_REDUX_STATE : {};

const Store = createStoreMiddleware(rootReducer, initialState, composeWithDevTools());

export { Store };

export default Store;
