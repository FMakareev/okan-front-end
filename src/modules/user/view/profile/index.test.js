import React from 'react';
import renderer from 'react-test-renderer';
import { Provider as ProviderRedux } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

import { ProfilePage } from './index';
import { StyledThemeProvider } from '../../../../styles/StyledThemeProvider';
import { Store } from '../../../../store';

test('ProfilePage: рендер без ошибок', () => {
  const output = renderer.create(
    <StyledThemeProvider>
      <BrowserRouter>
        <ProviderRedux store={Store}>
          <ProfilePage />
        </ProviderRedux>
      </BrowserRouter>
    </StyledThemeProvider>,
  );
  expect(output).toMatchSnapshot();
});
