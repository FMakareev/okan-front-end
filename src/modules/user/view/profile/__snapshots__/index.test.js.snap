// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfilePage: рендер без ошибок 1`] = `
<div
  className="sc-bxivhb dUiVDW"
>
  <div
    className="sc-1jxwge6-0 wterhv-0 hDVSSF"
  >
    <div
      className="sc-1jxwge6-0 wterhv-0 gZRpMZ"
    >
      <div
        className="sc-1jxwge6-0 wterhv-0 y44knq-0 fZsyTP"
      >
        <div
          className="sc-1jxwge6-0 sc-18ch56c-0 cfzSSs"
          color="color7"
          fontFamily="primary500"
          fontSize={6}
        >
          На согласование
        </div>
        <tbody
          className="obix2s-0 XmuzM"
        >
          <div
            className="sc-1jxwge6-0 wterhv-0 shCyQ"
          >
            <div
              className="sc-1jxwge6-0 sc-18ch56c-0 dXbZnM"
              color="color11"
              fontFamily="primary500"
              fontSize={6}
            >
              Номер документа
            </div>
            <div
              className="sc-1jxwge6-0 sc-18ch56c-0 dXbZnM"
              color="color11"
              fontFamily="primary500"
              fontSize={6}
            >
              Название документа
            </div>
          </div>
        </tbody>
        <tbody
          className="obix2s-0 sc-1eogwe1-1 cdJJct"
        >
          <tr
            className="sc-1cxzlk0-0 sc-1eogwe1-0 eHxsCS"
          >
            <td
              className="ivgi64-0 gbcbca"
              color="#333333"
              fontFamily="primary500"
              fontSize="18px"
            >
              23415
            </td>
            <td
              className="ivgi64-0 KGbJu"
              color="#333333"
              fontFamily="secondaryBold"
              fontSize="18px"
            >
              ТЗ - RK-186-344
            </td>
          </tr>
        </tbody>
        <tbody
          className="obix2s-0 sc-1eogwe1-1 cdJJct"
        >
          <tr
            className="sc-1cxzlk0-0 sc-1eogwe1-0 eHxsCS"
          >
            <td
              className="ivgi64-0 gbcbca"
              color="#333333"
              fontFamily="primary500"
              fontSize="18px"
            >
              23415
            </td>
            <td
              className="ivgi64-0 KGbJu"
              color="#333333"
              fontFamily="secondaryBold"
              fontSize="18px"
            >
              ТЗ - RK-186-344
            </td>
          </tr>
        </tbody>
      </div>
      <div
        className="sc-1jxwge6-0 wterhv-0 y44knq-1 gXqVMg"
      />
    </div>
    <div
      className="sc-1jxwge6-0 wterhv-0 jLxsDH"
    >
      <div
        className="sc-1jxwge6-0 wterhv-0 y44knq-0 fZsyTP"
      >
        <div
          className="sc-1jxwge6-0 sc-18ch56c-0 fNohKK"
          color="color7"
          fontFamily="primary500"
          fontSize={6}
        >
          Оповещения
        </div>
        <div
          className="sc-1jxwge6-0 sc-18ch56c-0 nokpo5-0 chxlbu"
          color="color11"
          fontFamily="secondary"
          fontSize={6}
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim
        </div>
        <div
          className="sc-1jxwge6-0 sc-18ch56c-0 nokpo5-0 chxlbu"
          color="color11"
          fontFamily="secondary"
          fontSize={6}
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim
        </div>
      </div>
      <div
        className="sc-1jxwge6-0 wterhv-0 y44knq-1 gXqVMg"
      />
    </div>
  </div>
</div>
`;
